web-services/query/src/main/java/datawave/webservice/query/runner/RunningQuery.java

```diff
@@ -173,6 +174,9 @@ public void setConnection(Connector connection) throws Exception {
             // TODO: applyPrediction("Plan");
         } catch (Exception e) {
             log.error(e.getMessage(), e);
+            if (this.logic instanceof BaseQueryLogic) {
```

Comment:
```suggestion
            if (this.logic instanceof BaseQueryLogic && ((BaseQueryLogic)this.logic).getCollectQueryMetrics()) {
```

Reply:
Not all BaseQueryLogics support metrics.

Reply:
Good catch!